go/pools/smartconnpool/pool_test.go

```diff
@@ -163,14 +163,16 @@ func TestOpen(t *testing.T) {
 		for i := 0; i < 5; i++ {
 			p.put(resources[i])
 		}
-		ch <- true
+		close(done)
 	}()
 	for i := 0; i < 5; i++ {
-		// Sleep to ensure the goroutine waits
-		time.Sleep(10 * time.Millisecond)
+		// block until we have a client wait for a connection, then offer it
+		for p.wait.waiting() == 0 {
+			time.Sleep(time.Millisecond)
+		}
```

Comment:
I feel like we should use a context here to break out of this if needed... but this is a unit test so the test itself should time out relatively quickly.